src/encoded/static/components/matrix_sescc_stem_cell.js

```diff
@@ -0,0 +1,676 @@
+
+                            if (element) {
+                                element.style.backgroundColor = d.selectedColor;
+                            }
+                        })
+                        .on('mouseout', (d) => {
+                            const name = formatH9HeaderTitle(d.name);
+                            const elementClass = formatPebbleNameToCssClassFriendly(name);
+                            const element = document.querySelector(`.${elementClass} th`);
+
+                            if (element) {
+                                element.style.backgroundColor = 'white';
+                            }
+                        });
+
+                    const ellipseSettings = [
+                        { cx: 2, cy: 2, rx: 12, ry: 9, stroke: 'black', 'stroke-width': 1.5 },
+                        { cx: 2, cy: 2, rx: 6.5, ry: 6, stroke: 'black', 'stroke-width': 1.5 },
+                        { cx: -5, cy: 10, rx: 12, ry: 9, stroke: 'black', 'stroke-width': 1.5 },
+                        { cx: -5, cy: 10, rx: 6.5, ry: 5, stroke: 'black', 'stroke-width': 1.5 },
+                        { cx: 10, cy: 10, rx: 12, ry: 9, stroke: 'black', 'stroke-width': 1.5 },
+                        { cx: 10, cy: 10, rx: 6.5, ry: 5, stroke: 'black', 'stroke-width': 1.5 },
+                    ];
+
+                    nodeEnter.append('svg:title').text(d => `Click to toggle matrix row: ${d.name}`);
+
+                    ellipseSettings.forEach((ellipseSetting) => {
+                        nodeEnter.append('ellipse')
+                            .attr('cx', ellipseSetting.cx)
+                            .attr('cy', ellipseSetting.cy)
+                            .attr('rx', ellipseSetting.rx)
+                            .attr('ry', ellipseSetting.ry)
+                            .style('stroke', ellipseSetting.stroke)
+                            .style('stroke-width', ellipseSetting['stroke-width'])
+                            .style('fill', d => d.selectedColor)
+                            .attr('class', () => 'js-cell');
+                    });
+
+                    nodeEnter.append('text')
+                        .attr('y', () => -18)
+                        .attr('dy', '.35em')
+                        .attr('text-anchor', 'middle')
+                        .attr('transform', 'rotate(310)')
+                        .text(d => d.name)
+                        .style('fill-opacity', 1);
+
+                    // Declare the link
+                    const link = svg.selectAll('path.sescc_matrix__link')
+                        .data(links, d => d.target.id);
+
+                    // Update the links.
+                    link.enter().insert('path', 'g')
+                        .attr('class', 'sescc_matrix__link')
+                        .style('stroke', d => d.target.linkColor)
+                        .attr('d', diagonal);
+                };
+                update(treeData[0]);
+            });
+        }
+    }
```

Comment:
Oh wow; I thought this was a static SVG. I didn’t know you were rendering it in D3. Do you think this is better than using a static image? I just worry this is a lot more difficult to maintain than having an SVG you can modify in Illustrator. I don’t _think_ the shape of this SVG changes except for CSS classes.

Reply:
Are you okay with me leaving it as-is? The issue is that the SVG code can get messy wiring up the events and handing the styles if embedded; and can be a sore to look at if inline. Plus, the SVG has some dynamic changes like the height and width is adjusted at runtime if the screen changes. If you feel strongly against D3.js, I will look into it.

Reply:
No, I would recommend keeping it this way; I was just surprised. And maybe others would prefer the way you did it. An embedded or file-based SVG should be able to handle height and width changes on window-size changes though, generally at least. I don’t know if you have a complicated case.